lib/tree.rb

```diff
@@ -16,45 +16,124 @@ def initialize
     @root = nil
   end
 
-  # Time Complexity: 
-  # Space Complexity: 
-  def add(key, value)
-    raise NotImplementedError
+  # Time Complexity: O(log n) 
+  # Space Complexity: O(log n)
+  def add(key, value = nil)
```

Comment:
👍 Good use of default parameters.  The space/time complexities are O(n) if the tree is unbalanced and O(log n) if the tree is balanced.